### DISCUSSION TITLE
Feature request: add `dom.unwatch()` or similar

### DISCUSSION THREAD

**Discussion by pablote** [FortAwesome/Font-Awesome#20922] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Yes, a performance problem. I'm using FontAwesome SVG core, selecting which fonts I need to use and calling `dom.watch()`. 

This works great most of the time, except for a specific section of my page which generates a lot of DOM dynamically, several list items each with a few icons themselves. This takes around 3s/4s on my browser, all this time taken by Font Awesome, it'd be instantaneous otherwise.

**Describe the solution you'd like**
I noticed that if I don't call `dom.watch()` and instead just call `dom.i2svg()` after my DOM operations are done, everything actually works pretty fast. Of course I can't just do this because it breaks other usages of Font Awesome throughout the site. What I would like instead is, being able to disable `dom.watch()` when I know I'm about to do some DOM heavy stuff (maybe call it `dom.unwatch()` ?) and then enable it back again.

**Describe alternatives you've considered**
I considered not using `dom.watch()` at all. Although this requires some development effort, mainly call `dom.i2svg()` everytime I add a component to the page that used Font Awesome.

**Feature request checklist**
- [x] This is a single feature (i.e. not a re-write of all of Font Awesome)
- [x] The title starts with "Feature request: " and is followed by a clear feature name (Ex: `Feature request: moar cowbell`)
- [x] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome-Pro/issues) and to the best of my knowledge this is not a duplicate

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community.

Looks an interesting feature to have, do you have a suggestion for a PR?

@robmadole is there a plan to include this feature in the future? I've had a look to how `watch` is implemented but I'm not able to estimate the effort to make its complementary version

**Comment by robmadole** [comment]
@pablote thanks for the suggestion! That's a pretty good idea. I will add this to our version 6.x roadmap!

**Comment by ssaar** [comment]
Was this ever addressed, we're running in to a lot of perf issues with FA 6.5.2 whenever we have to add a ton of dom elements. What I would like is either the ability to ignore a section of dom, or at least temp unwatch, then watch the dom again. Right now the peformance is killing us and we don't have any way to fix it. From looking at perf in chrome debugger it looks like this is the hot section of code:

mo = new MUTATION_OBSERVER(function (objects) {
      if (disabled) return;
      var defaultPrefix = getDefaultUsablePrefix();
      toArray(objects).forEach(function (mutationRecord) {
        if (mutationRecord.type === 'childList' && mutationRecord.addedNodes.length > 0 && !isWatched(mutationRecord.addedNodes[0])) {
          if (config.searchPseudoElements) {
            pseudoElementsCallback(mutationRecord.target);
          }

          treeCallback(mutationRecord.target);
        }

        if (mutationRecord.type === 'attributes' && mutationRecord.target.parentNode && config.searchPseudoElements) {
          pseudoElementsCallback(mutationRecord.target.parentNode);
        }

        if (mutationRecord.type === 'attributes' && isWatched(mutationRecord.target) && ~ATTRIBUTES_WATCHED_FOR_MUTATION.indexOf(mutationRecord.attributeName)) {
          if (mutationRecord.attributeName === 'class' && hasPrefixAndIcon(mutationRecord.target)) {
            var _getCanonicalIcon = getCanonicalIcon(classArray(mutationRecord.target)),
                prefix = _getCanonicalIcon.prefix,
                iconName = _getCanonicalIcon.iconName;

            mutationRecord.target.setAttribute(DATA_PREFIX, prefix || defaultPrefix);
            if (iconName) mutationRecord.target.setAttribute(DATA_ICON, iconName);
          } else if (hasBeenReplaced(mutationRecord.target)) {
            nodeCallback(mutationRecord.target);
          }
        }
      });

and perf graph looks like this:

<img width="1087" alt="image" src="https://github.com/user-attachments/assets/cbfd8da4-fb98-4876-a3b9-eb1ae7264888">

With the anonymous section being the code included above.
